lib/secure_headers/headers/cookie.rb

```diff
@@ -0,0 +1,187 @@
+            end
+          end
+        end
+
+        if config[:samesite]
+          raise CookiesConfigError.new("samesite cookie config must be a hash") unless config[:samesite].is_a?(Hash)
+
+          # when configuring with booleans, only one enforcement is permitted
+          if config[:samesite].key?(:lax) && config[:samesite][:lax].is_a?(TrueClass) && config[:samesite].key?(:strict)
+            raise CookiesConfigError.new("samesite cookie config is invalid, combination use of booleans and Hash to configure lax and strict enforcement is not permitted.")
+          elsif config[:samesite].key?(:strict) && config[:samesite][:strict].is_a?(TrueClass) && config[:samesite].key?(:lax)
+            raise CookiesConfigError.new("samesite cookie config is invalid, combination use of booleans and Hash to configure lax and strict enforcement is not permitted.")
+          end
+
+          # validate Hash-based samesite configuration
+          if config[:samesite].key?(:lax) && config[:samesite][:lax].is_a?(Hash)
+            # validate exclusive use of only or except but not both at the same time
+            if config[:samesite][:lax].key?(:only) && config[:samesite][:lax].key?(:except)
+              raise CookiesConfigError.new("samesite lax cookie config is invalid, simultaneous use of conditional arguments `only` and `except` is not permitted.")
+            end
+
+            if config[:samesite].key?(:strict)
+              # validate exclusivity of only and except members
+              if (intersection = (config[:samesite][:lax].fetch(:only, []) & config[:samesite][:strict].fetch(:only, []))).any?
+                raise CookiesConfigError.new("samesite cookie config is invalid, cookie(s) #{intersection.join(', ')} cannot be enforced as lax and strict")
+              end
+
+              if (intersection = (config[:samesite][:lax].fetch(:except, []) & config[:samesite][:strict].fetch(:except, []))).any?
+                raise CookiesConfigError.new("samesite cookie config is invalid, cookie(s) #{intersection.join(', ')} cannot be enforced as lax and strict")
+              end
+            end
+          end
+
+          if config[:samesite].key?(:strict) && config[:samesite][:strict].is_a?(Hash)
+            # validate exclusive use of only or except but not both at the same time
+            if config[:samesite][:strict].key?(:only) && config[:samesite][:strict].key?(:except)
+              raise CookiesConfigError.new("samesite strict cookie config is invalid, simultaneous use of conditional arguments `only` and `except` is not permitted.")
+            end
+          end
+        end
+      end
+    end
+
+    attr_reader :raw_cookie, :config
+
+    def initialize(cookie, config)
+      @raw_cookie = cookie
+      @config = config
+      @attributes = {
+        "secure" => nil,
```

Comment:
We can certainly do that, my reasoning for using strings was that parsing the cookie in order to decompose it would require `to_sym` on strings in order to set these to a truthy value. I guess now that Ruby garbage collects symbols this is less of a concern.